File: TwoPointers/167.TwoSumII.java
// https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int result[] = new int[2];
        int i=0, j=numbers.length-1;
        while (i != j) {
            int sum = numbers[i] + numbers[j];
            if (sum == target) {
                result[0] = i+1;
                result[1] = j+1;  
                break;
            } else if (sum > target) {
                j--;
            } else {
                i++;
            }
        }
        return result;
    }
}

File: TwoPointers/167.TwoSumII.swift
// https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/
class Solution {
    func twoSum(_ numbers: [Int], _ target: Int) -> [Int] {
        var result:[Int] = []
        let n = numbers.count
        var l = 0
        var r = n - 1
        while (l != r) {
            var sum = numbers[l] + numbers[r]
            if (sum == target) {
                result.append(l+1)
                result.append(r+1)
                break
            } else if (sum < target) {
                l = l + 1
            } else {
                r = r - 1
            }
        }
        return result
    }
}
